### ISSUE TITLE
Complain expected when assigning Promise<A | B> to Promise<A>

### ISSUE THREAD

**Issue by dggluz** [Microsoft/TypeScript#16202]
**TypeScript Version:**  2.2.1 / nightly (2.2.0-dev.201xxxxx)

**Code**

```ts
// This fails as number is not assignable to string
let strOrNum: string | number;
let str: string = strOrNum;

// This fails for the same reason
let strOrNumA: Array<string | number>
let strA: Array<string> = strOrNumA;

// This works, and I think it shouldn't
let strOrNumP: Promise<string | number>
let strP: Promise<string> = strOrNumP;
```

**Expected behavior:** 
Expected TypeScript to complain about the assignment of `strP` (as it does with `strA`).

**Actual behavior:** 
TypeScript doesn't complain.

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/11022, fixed by https://github.com/Microsoft/TypeScript/pull/15104. should be working as expected in TS 2.4
